src/builtins/core/instant.rs

```diff
@@ -157,16 +157,16 @@ impl Instant {
 impl Instant {
     /// Create a new validated `Instant`.
     #[inline]
-    pub fn try_new(nanoseconds: i128) -> TemporalResult<Self> {
-        Ok(Self::from(EpochNanoseconds::try_from(nanoseconds)?))
+    pub fn try_new(nanoseconds: EpochNanoseconds) -> TemporalResult<Self> {
+        Ok(Self::from(EpochNanoseconds::try_from(nanoseconds).unwrap()))
     }
 
     /// Creates a new `Instant` from the provided Epoch millisecond value.
-    pub fn from_epoch_milliseconds(epoch_milliseconds: i64) -> TemporalResult<Self> {
+    pub fn from_epoch_milliseconds(epoch_milliseconds: EpochNanoseconds) -> TemporalResult<Self> {
```

Comment:
Hmmmmm, I believe this isn't entirely correct since we are dealing with milliseconds vs. nanoseconds. This should probably be left as is or a new `EpochMilliseconds` primitive type could be introduces with an internal representation of `i64`.